pkg/engine/event/combat.go

```diff
@@ -15,6 +18,20 @@ type AttackStartEvent struct {
 	DamageType model.DamageType
 }
 
+func (e AttackStartEvent) MarshalJSON() ([]byte, error) {
+	var buf bytes.Buffer
+	buf.WriteString("{\"Event\": \"AttackStartEvent\",")
+	buf.WriteString("\"Attacker\": " + strconv.Itoa(int(e.Attacker)))
+	buf.WriteString(fmt.Sprintf(",\"Targets\": %v", e.Targets))
+	buf.WriteString(",\"AttackType\": \"" + e.AttackType.String())
+	buf.WriteString("\",\"DamageType\": \"" + e.DamageType.String())
+	buf.WriteByte('"')
+	buf.WriteByte('}')
+	fmt.Println("Hello")
+	fmt.Println(buf.String())
+	return buf.Bytes(), nil
+}
```

Comment:
from offline discussion, ideally do not want to create a custom `MarshalJSON` for every event. Instead:
1. `logger` should have some generic way to wrap events and add the name
2. rely on nested `json.Marshal` calls rather than modifying bytes directly like here

IE can do something like the following:
```go
type log struct {
  event string      `json:event`
  data  interface{} `json:data`
}

func (l *logger) Log(e interface{}) {
  json.Marshal(log{
    event: <some way to get its name>,
    data:  e,
  })
}
```

By design, `json.Marshal` will call `json.Marshal` on each field value of a struct (marshalling is recursive)